scripts/common/timer.ts

```diff
@@ -162,11 +161,37 @@ export function generateOverallComparisonSplit(baseRun: RunMetadata, comparisonR
 	};
 }
 
-export function generateComparisonSplits(baseRunSplits: RunSplits, comparisonRunSplits: RunSplits): ComparisonSplit[] {
-	return baseRunSplits.segments.map((_, i) =>
-		// TODO: Generate subsegment splits
-		generateSegmentComparisonSplit(baseRunSplits, comparisonRunSplits, i, 0)
-	);
+export function generateComparisonSplits(baseRun: RunMetadata, comparisonRun: RunMetadata): ComparisonSplit[][] {
+	const comparisonSplits: ComparisonSplit[][] = [];
+	for (let i = 0; i < baseRun.runSplits.segments.length; i++) {
+		for (let j = 0; j < baseRun.runSplits.segments[i].subsegments.length; j++) {
+			if (i === 0 && j === 0) {
+				// Skip first split
+				continue;
+			}
+
+			const splitIndex = Math.max(i - 1, 0);
+			if (comparisonSplits.length <= splitIndex) {
+				comparisonSplits.push([]);
+			}
+
+			comparisonSplits[splitIndex].push(
+				generateSegmentComparisonSplit(baseRun.runSplits, comparisonRun.runSplits, i, j)
+			);
+		}
+	}
+
+	const finishSplit = [
+		generateFinishSplitComparison(
+			baseRun.runTime,
+			baseRun.runSplits,
+			comparisonRun.runTime,
+			comparisonRun.runSplits
+		)
+	];
+	comparisonSplits.push(finishSplit);
+
+	return comparisonSplits;
```

Comment:
Untested!
```ts
export function generateComparisonSplits(baseRun: RunMetadata, comparisonRun: RunMetadata): ComparisonSplit[][] {
	const comparisonSplits = baseRun.runSplits.segments.map((segment, i) =>
		segment.subsegments
			// Skip first split
			.filter((_, j) => i !== 0 && j !== 0)
			.map((_, j) => generateSegmentComparisonSplit(baseRun.runSplits, comparisonRun.runSplits, i, j))
	);

	comparisonSplits.push([
		generateFinishSplitComparison(
			baseRun.runTime,
			baseRun.runSplits,
			comparisonRun.runTime,
			comparisonRun.runSplits
		)
	]);

	return comparisonSplits;
}
```

Reply:
generateSegmentComparisonSplit(baseRun.runSplits, comparisonRun.runSplits, i, i === 0 ? j - 1 : j)

Reply:
I can fix your code by changing the filter condition to `!(i === 0 && j === 0)` and adding a `.slice(1)` after the `.map(...)`. I don't think this makes things more readable compared to the for loop though. Using a `.slice()` to account for the empty array is very unintuitive.